<!DOCTYPE html>
<html  lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1"><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

      <title>NIH/3T3 (seqFISH+)</title>
    
          <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
          <link rel="stylesheet" href="../_static/theme.css " type="text/css" />
          <link rel="stylesheet" href="../_static/sg_gallery.css" type="text/css" />
          <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
      
      <!-- sphinx script_files -->
        <script src="../_static/documentation_options.js?v=8d563738"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>

      
      <!-- bundled in js (rollup iife) -->
      <!-- <script src="../_static/theme-vendors.js"></script> -->
      <script src="../_static/theme.js" defer></script>
    
  <link rel="index" title="Index" href="../genindex.html" />
  <link rel="search" title="Search" href="../search.html" />
  <link rel="next" title="Release notes" href="../release_notes/index.html" />
  <link rel="prev" title="Mouse olfactory bulb (Stereo-seq)" href="Stereo-seq.html" /> 
  </head>

  <body>
    <div id="app">
    <div class="theme-container" :class="pageClasses"><navbar @toggle-sidebar="toggleSidebar">
  <router-link to="../index.html" class="home-link">
    
      <img class="logo" src="../../_static/Cellist_logo_blue_px.png" alt="logo"/>
    
  </router-link>

  <div class="links">
    <navlinks class="can-hide">

  
    <div class="nav-item">
      <a href="../index.html#welcome-to-cellist-s-documentation"
         class="nav-link ">
         Contents:
      </a>
    </div>
  
    <div class="nav-item">
      <a href="../index.html#id1"
         class="nav-link ">
         1.0.0 2024-04-16
      </a>
    </div>
  



    </navlinks>
  </div>
</navbar>

      
      <div class="sidebar-mask" @click="toggleSidebar(false)">
      </div>
        <sidebar @toggle-sidebar="toggleSidebar">
          
          <navlinks>
            

  
    <div class="nav-item">
      <a href="../index.html#welcome-to-cellist-s-documentation"
         class="nav-link ">
         Contents:
      </a>
    </div>
  
    <div class="nav-item">
      <a href="../index.html#id1"
         class="nav-link ">
         1.0.0 2024-04-16
      </a>
    </div>
  



            
          </navlinks><div id="searchbox" class="searchbox" role="search">
  <div class="caption"><span class="caption-text">Quick search</span>
    <div class="searchformwrapper">
      <form class="search" action="../search.html" method="get">
        <input type="text" name="q" />
        <input type="submit" value="Search" />
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div><div class="sidebar-links" role="navigation" aria-label="main navigation">
  
    <div class="sidebar-group">
      <p class="caption">
        <span class="caption-text"><a href="../index.html#welcome-to-cellist-s-documentation">Contents:</a></span>
      </p>
      <ul class="">
        
      </ul>
    </div>
  
    <div class="sidebar-group">
      <p class="caption">
        <span class="caption-text"><a href="../index.html#id1">1.0.0 2024-04-16</a></span>
      </p>
      <ul class="">
        
          <li class="toctree-l1 ">
            
              <a href="../installation.html" class="reference internal ">Installation</a>
            

            
          </li>

        
          <li class="toctree-l1 ">
            
              <a href="../tutorials.html" class="reference internal ">Tutorials</a>
            

            
          </li>

        
          <li class="toctree-l1 ">
            
              <a href="../release_notes/index.html" class="reference internal ">Release notes</a>
            

            
          </li>

        
      </ul>
    </div>
  
</div>
        </sidebar>

      <page>
          <div class="body-header" role="navigation" aria-label="navigation">
  
  <ul class="breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
    
      <li><a href="../tutorials.html">Tutorials</a> &raquo;</li>
    
    <li>NIH/3T3 (seqFISH+)</li>
  </ul>
  

  <ul class="page-nav">
  <li class="prev">
    <a href="Stereo-seq.html"
       title="previous chapter">← Mouse olfactory bulb (Stereo-seq)</a>
  </li>
  <li class="next">
    <a href="../release_notes/index.html"
       title="next chapter">Release notes →</a>
  </li>
</ul>
  
</div>
<hr>
          <div class="content" role="main" v-pre>
            
  <section id="nih-3t3-seqfish">
<h1>NIH/3T3 (seqFISH+)<a class="headerlink" href="#nih-3t3-seqfish" title="Link to this heading">¶</a></h1>
<p>Here we demonstrate the application of Cellist on the imaging-based high-resolution platform. The NIH/3T3 seqFISH+ data was derived from the study of <a class="reference external" href="https://www.nature.com/articles/s41586-019-1049-y">Eng et al., Nature, 2019</a>, which profiles 10,000 genes in 7 field of views (FOVs) with auxiliary DAPI staining. Users can download the demo data from <a class="reference external" href="https://github.com/wanglabtongji/Cellist/tree/main/test/seqFISH_NIH3T3">here</a>.</p>
<section id="step-1-pre-process">
<h2>Step 1 Pre-process<a class="headerlink" href="#step-1-pre-process" title="Link to this heading">¶</a></h2>
<p>Cellist is initially designed for the barcoding-based high-resolution spatial transcriptomics where each spot captures multiple transcripts. But in the seqFISH+ data of single-molecule resolution, each pixel is around 103 nm. To make the seqFISH+ data suitable for Cellist, transcripts are aggregated within each bin (5 pixels × 5 pixels), treating each bin as a spot akin to Stereo-seq. This process results in binned data with a comparable resolution (~0.5 μm) to Stereo-seq, suitable for Cellist.</p>
<section id="convert-array-like-data-to-long-data-frame">
<h3>1. Convert array-like data to long data frame<a class="headerlink" href="#convert-array-like-data-to-long-data-frame" title="Link to this heading">¶</a></h3>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">scipy.io</span> <span class="k">as</span> <span class="nn">sio</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">Cellist.Utility</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">Cellist.Plot</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">from</span> <span class="nn">Cellist.IO</span> <span class="kn">import</span> <span class="o">*</span>

<span class="n">data_dir</span> <span class="o">=</span> <span class="s1">&#39;Data&#39;</span>
<span class="n">res_dir</span> <span class="o">=</span> <span class="s1">&#39;Result/Preprocess&#39;</span>
<span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">res_dir</span><span class="p">):</span>
   <span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span><span class="n">res_dir</span><span class="p">)</span>

<span class="n">run_mat_file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span> <span class="s1">&#39;RNA_locations_run_1.mat&#39;</span><span class="p">)</span>
<span class="n">gene_mat_file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span> <span class="s1">&#39;all_gene_Names.mat&#39;</span><span class="p">)</span>

<span class="n">mat_contents</span> <span class="o">=</span> <span class="n">sio</span><span class="o">.</span><span class="n">loadmat</span><span class="p">(</span><span class="n">run_mat_file</span><span class="p">)</span>
<span class="n">tot</span> <span class="o">=</span> <span class="n">mat_contents</span><span class="p">[</span><span class="s1">&#39;tot&#39;</span><span class="p">]</span>

<span class="n">gene_mat</span> <span class="o">=</span> <span class="n">sio</span><span class="o">.</span><span class="n">loadmat</span><span class="p">(</span><span class="n">gene_mat_file</span><span class="p">)</span>
<span class="n">gene_list_all</span> <span class="o">=</span> <span class="n">gene_mat</span><span class="p">[</span><span class="s1">&#39;allNames&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span>
<span class="n">gene_list_all</span> <span class="o">=</span> <span class="p">[</span><span class="n">i</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">gene_list_all</span><span class="p">]</span>

<span class="n">x_list</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">y_list</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">cell_list</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">gene_list</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">fov_list</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">tot</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]):</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">i</span><span class="p">)</span>
    <span class="k">for</span> <span class="n">j</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">tot</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]):</span>
        <span class="k">for</span> <span class="n">m</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">tot</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">2</span><span class="p">]):</span>
            <span class="k">if</span> <span class="n">tot</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="n">j</span><span class="p">,</span><span class="n">m</span><span class="p">]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="o">==</span> <span class="mi">3</span><span class="p">:</span>
                <span class="n">x_list</span> <span class="o">=</span> <span class="n">x_list</span> <span class="o">+</span> <span class="n">tot</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="n">j</span><span class="p">,</span><span class="n">m</span><span class="p">][:,</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
                <span class="n">y_list</span> <span class="o">=</span> <span class="n">y_list</span> <span class="o">+</span> <span class="n">tot</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="n">j</span><span class="p">,</span><span class="n">m</span><span class="p">][:,</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
                <span class="n">cell_list</span> <span class="o">=</span> <span class="n">cell_list</span> <span class="o">+</span> <span class="p">[</span><span class="s1">&#39;Cell_</span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span><span class="n">j</span><span class="p">]</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">tot</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="n">j</span><span class="p">,</span><span class="n">m</span><span class="p">][:,</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
                <span class="n">gene_list</span> <span class="o">=</span> <span class="n">gene_list</span> <span class="o">+</span> <span class="p">[</span><span class="n">gene_list_all</span><span class="p">[</span><span class="n">m</span><span class="p">]]</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">tot</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="n">j</span><span class="p">,</span><span class="n">m</span><span class="p">][:,</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
                <span class="n">fov_list</span> <span class="o">=</span> <span class="n">fov_list</span> <span class="o">+</span> <span class="p">[</span><span class="s1">&#39;FOV_</span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span><span class="n">i</span><span class="p">]</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">tot</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="n">j</span><span class="p">,</span><span class="n">m</span><span class="p">][:,</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>

<span class="n">coord_count_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s1">&#39;x&#39;</span><span class="p">:</span><span class="n">x_list</span><span class="p">,</span> <span class="s1">&#39;y&#39;</span><span class="p">:</span><span class="n">y_list</span><span class="p">,</span> <span class="s1">&#39;Manual_cell&#39;</span><span class="p">:</span> <span class="n">cell_list</span><span class="p">,</span> <span class="s1">&#39;geneID&#39;</span><span class="p">:</span> <span class="n">coord_count_df</span><span class="p">[</span><span class="s1">&#39;geneID&#39;</span><span class="p">],</span> <span class="s1">&#39;FOV&#39;</span><span class="p">:</span> <span class="n">fov_list</span><span class="p">,</span> <span class="s1">&#39;MIDCounts&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">})</span>
<span class="n">coord_count_df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">res_dir</span><span class="p">,</span> <span class="s1">&#39;seqFISH+_NIH3T3_points_RNA_rep1.txt&#39;</span><span class="p">),</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="generate-pixel-level-and-bin-level-data-5-pixel-bin">
<h3>2. Generate pixel-level and bin-level data (5-pixel bin)<a class="headerlink" href="#generate-pixel-level-and-bin-level-data-5-pixel-bin" title="Link to this heading">¶</a></h3>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">coord_count_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">res_dir</span><span class="p">,</span> <span class="s1">&#39;seqFISH+_NIH3T3_points_RNA_rep1.txt&#39;</span><span class="p">),</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>
<span class="k">for</span> <span class="n">fov</span> <span class="ow">in</span> <span class="n">coord_count_df</span><span class="p">[</span><span class="s1">&#39;FOV&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
   <span class="nb">print</span><span class="p">(</span><span class="n">fov</span><span class="p">)</span>
   <span class="n">out_dir</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">res_dir</span><span class="p">,</span> <span class="n">fov</span><span class="p">)</span>
   <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">out_dir</span><span class="p">):</span>
      <span class="n">os</span><span class="o">.</span><span class="n">mkdir</span><span class="p">(</span><span class="n">out_dir</span><span class="p">)</span>
   <span class="n">fov_count_df</span> <span class="o">=</span> <span class="n">coord_count_df</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
   <span class="n">fov_count_df</span> <span class="o">=</span> <span class="n">fov_count_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">fov_count_df</span><span class="p">[</span><span class="s1">&#39;FOV&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">fov</span><span class="p">,:]</span>
   <span class="n">fov_count_df</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">fov_count_df</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">int</span><span class="p">)</span>
   <span class="n">fov_count_df</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">fov_count_df</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">int</span><span class="p">)</span>
   <span class="n">fov_coord_df</span> <span class="o">=</span> <span class="n">fov_count_df</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">([</span><span class="s2">&quot;x&quot;</span><span class="p">,</span> <span class="s2">&quot;y&quot;</span><span class="p">])</span>
   <span class="n">fov_coord_df</span><span class="p">[</span><span class="s1">&#39;Manual_cell_ID&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">fov_coord_df</span><span class="p">[</span><span class="s1">&#39;Manual_cell&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="s1">&#39;category&#39;</span><span class="p">)</span>
   <span class="n">fov_coord_df</span><span class="p">[</span><span class="s1">&#39;Manual_cell_ID&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">fov_coord_df</span><span class="p">[</span><span class="s1">&#39;Manual_cell_ID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">cat</span><span class="o">.</span><span class="n">codes</span> <span class="o">+</span> <span class="mi">1</span>
   <span class="c1"># draw the manual segmentation result</span>
   <span class="n">draw_segmentation</span><span class="p">(</span><span class="n">fov_coord_df</span><span class="p">,</span> <span class="s2">&quot;Manual_cell_ID&quot;</span><span class="p">,</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">_Manual&quot;</span> <span class="o">%</span><span class="n">fov</span><span class="p">,</span> <span class="n">out_dir</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span> <span class="s2">&quot;x&quot;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s2">&quot;y&quot;</span><span class="p">,</span> <span class="n">figsize</span> <span class="o">=</span> <span class="p">(</span><span class="mi">20</span><span class="p">,</span><span class="mi">20</span><span class="p">))</span>
   <span class="c1"># write bin1</span>
   <span class="n">bin1_count_df</span> <span class="o">=</span> <span class="n">coord_count_df</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
   <span class="n">bin1_count_df</span> <span class="o">=</span> <span class="n">bin1_count_df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;FOV&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">fov</span><span class="p">,:]</span>
   <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">int</span><span class="p">)</span>
   <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">int</span><span class="p">)</span>
   <span class="n">bin1_count_df</span> <span class="o">=</span> <span class="n">bin1_count_df</span><span class="p">[[</span><span class="s2">&quot;geneID&quot;</span><span class="p">,</span> <span class="s2">&quot;x&quot;</span><span class="p">,</span> <span class="s2">&quot;y&quot;</span><span class="p">,</span> <span class="s2">&quot;MIDCounts&quot;</span><span class="p">]]</span>
   <span class="n">bin1_count_df</span> <span class="o">=</span> <span class="n">bin1_count_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;x&quot;</span><span class="p">,</span> <span class="s2">&quot;y&quot;</span><span class="p">])</span>
   <span class="n">bin1_count_df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">out_dir</span><span class="p">,</span> <span class="s1">&#39;seqFISH+_NIH3T3_points_RNA_rep1_</span><span class="si">%s</span><span class="s1">_bin1.txt&#39;</span> <span class="o">%</span><span class="n">fov</span><span class="p">),</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
   <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;x_y&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span> <span class="o">+</span> <span class="s1">&#39;_&#39;</span> <span class="o">+</span> <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span>
   <span class="n">gene_spot</span> <span class="o">=</span> <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;MIDCounts&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;x_y&#39;</span><span class="p">],</span> <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;geneID&#39;</span><span class="p">]])</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
   <span class="n">spot_expr_mat</span><span class="p">,</span> <span class="n">gene_list</span><span class="p">,</span> <span class="n">spot_list</span> <span class="o">=</span> <span class="n">longdf_to_mat</span><span class="p">(</span><span class="n">gene_spot</span><span class="p">)</span>
   <span class="n">count_h5_file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">out_dir</span><span class="p">,</span> <span class="s2">&quot;seqFISH+_NIH3T3_points_RNA_rep1_</span><span class="si">%s</span><span class="s2">_bin1.h5&quot;</span> <span class="o">%</span><span class="n">fov</span><span class="p">)</span>
   <span class="n">write_10X_h5</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="n">count_h5_file</span><span class="p">,</span> <span class="n">matrix</span> <span class="o">=</span> <span class="n">spot_expr_mat</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">gene_list</span><span class="p">,</span> <span class="n">barcodes</span> <span class="o">=</span> <span class="n">spot_list</span><span class="p">,</span> <span class="n">datatype</span> <span class="o">=</span> <span class="s1">&#39;Gene&#39;</span><span class="p">)</span>
   <span class="c1"># write bin5</span>
   <span class="n">countname</span> <span class="o">=</span> <span class="s1">&#39;MIDCounts&#39;</span>
   <span class="n">bin_size</span> <span class="o">=</span> <span class="mi">5</span>
   <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;x_bin&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">]</span><span class="o">/</span><span class="n">bin_size</span><span class="p">)</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">uint32</span><span class="p">)</span><span class="o">*</span><span class="n">bin_size</span>
   <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;y_bin&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">]</span><span class="o">/</span><span class="n">bin_size</span><span class="p">)</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">uint32</span><span class="p">)</span><span class="o">*</span><span class="n">bin_size</span>
   <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;binID&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;x_bin&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span> <span class="o">+</span> <span class="s2">&quot;_&quot;</span> <span class="o">+</span> <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;y_bin&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span>
   <span class="n">bin5_count_df</span> <span class="o">=</span> <span class="n">bin1_count_df</span><span class="p">[</span><span class="n">countname</span><span class="p">]</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;x_bin&#39;</span><span class="p">],</span> <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;y_bin&#39;</span><span class="p">],</span> <span class="n">bin1_count_df</span><span class="p">[</span><span class="s1">&#39;geneID&#39;</span><span class="p">]])</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
   <span class="n">bin5_count_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">bin5_count_df</span><span class="p">)</span>
   <span class="n">bin5_count_df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">,</span> <span class="s1">&#39;y&#39;</span><span class="p">,</span> <span class="s1">&#39;geneID&#39;</span><span class="p">]</span>
   <span class="n">bin5_count_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">bin5_count_df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">to_frame</span><span class="p">(),</span> <span class="n">bin5_count_df</span><span class="p">],</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
   <span class="n">bin5_count_df</span> <span class="o">=</span> <span class="n">bin5_count_df</span><span class="p">[[</span><span class="s1">&#39;geneID&#39;</span><span class="p">,</span> <span class="s1">&#39;x&#39;</span><span class="p">,</span> <span class="s1">&#39;y&#39;</span><span class="p">,</span> <span class="s1">&#39;MIDCounts&#39;</span><span class="p">]]</span>
   <span class="n">bin5_count_df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">out_dir</span><span class="p">,</span> <span class="s1">&#39;seqFISH+_NIH3T3_points_RNA_rep1_</span><span class="si">%s</span><span class="s1">_bin5.txt&#39;</span> <span class="o">%</span><span class="n">fov</span><span class="p">),</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
   <span class="n">bin5_count_df</span><span class="p">[</span><span class="s1">&#39;x_y&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">bin5_count_df</span><span class="p">[</span><span class="s1">&#39;x&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span> <span class="o">+</span> <span class="s1">&#39;_&#39;</span> <span class="o">+</span> <span class="n">bin5_count_df</span><span class="p">[</span><span class="s1">&#39;y&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span>
   <span class="n">gene_spot</span> <span class="o">=</span> <span class="n">bin5_count_df</span><span class="p">[</span><span class="s1">&#39;MIDCounts&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="n">bin5_count_df</span><span class="p">[</span><span class="s1">&#39;x_y&#39;</span><span class="p">],</span> <span class="n">bin5_count_df</span><span class="p">[</span><span class="s1">&#39;geneID&#39;</span><span class="p">]])</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
   <span class="n">spot_expr_mat</span><span class="p">,</span> <span class="n">gene_list</span><span class="p">,</span> <span class="n">spot_list</span> <span class="o">=</span> <span class="n">longdf_to_mat</span><span class="p">(</span><span class="n">gene_spot</span><span class="p">)</span>
   <span class="n">count_h5_file</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">out_dir</span><span class="p">,</span>  <span class="s2">&quot;seqFISH+_NIH3T3_points_RNA_rep1_</span><span class="si">%s</span><span class="s2">_bin5.h5&quot;</span> <span class="o">%</span><span class="n">fov</span><span class="p">)</span>
   <span class="n">write_10X_h5</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="n">count_h5_file</span><span class="p">,</span> <span class="n">matrix</span> <span class="o">=</span> <span class="n">spot_expr_mat</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">gene_list</span><span class="p">,</span> <span class="n">barcodes</span> <span class="o">=</span> <span class="n">spot_list</span><span class="p">,</span> <span class="n">datatype</span> <span class="o">=</span> <span class="s1">&#39;Gene&#39;</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="convert-ome-tiff-file-to-2d-tiff-image">
<h3>3. Convert OME-TIFF file to 2D TIFF image<a class="headerlink" href="#convert-ome-tiff-file-to-2d-tiff-image" title="Link to this heading">¶</a></h3>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">pyometiff</span> <span class="kn">import</span> <span class="n">OMETIFFReader</span>
<span class="kn">from</span> <span class="nn">skimage.io</span> <span class="kn">import</span> <span class="n">imread</span><span class="p">,</span> <span class="n">imsave</span>

<span class="n">data_dir</span> <span class="o">=</span> <span class="s1">&#39;Data&#39;</span>
<span class="n">res_dir</span> <span class="o">=</span> <span class="s1">&#39;Result/Preprocess&#39;</span>

<span class="k">for</span> <span class="n">fov</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">7</span><span class="p">):</span>
   <span class="n">out_dir</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">res_dir</span><span class="p">,</span> <span class="s1">&#39;FOV_</span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span><span class="n">fov</span><span class="p">)</span>
   <span class="nb">print</span><span class="p">(</span><span class="n">out_dir</span><span class="p">)</span>
   <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">out_dir</span><span class="p">):</span>
     <span class="n">os</span><span class="o">.</span><span class="n">mkdir</span><span class="p">(</span><span class="n">out_dir</span><span class="p">)</span>
   <span class="n">img_path</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">data_dir</span><span class="p">,</span> <span class="s1">&#39;DAPI_experiment1/final_background_experiment1/MMStack_Pos</span><span class="si">%s</span><span class="s1">.ome.tif&#39;</span> <span class="o">%</span><span class="n">fov</span><span class="p">)</span>
   <span class="n">reader</span> <span class="o">=</span> <span class="n">OMETIFFReader</span><span class="p">(</span><span class="n">fpath</span> <span class="o">=</span> <span class="n">img_path</span><span class="p">)</span>
   <span class="n">img_array</span><span class="p">,</span> <span class="n">xml_metadata</span> <span class="o">=</span> <span class="n">reader</span><span class="o">.</span><span class="n">read</span><span class="p">()</span>
   <span class="n">img</span> <span class="o">=</span> <span class="n">img_array</span><span class="p">[</span><span class="mi">3</span><span class="p">,</span><span class="mi">1</span><span class="p">,:,:]</span>
   <span class="n">imsave</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">out_dir</span><span class="p">,</span> <span class="s1">&#39;MMStack_Pos</span><span class="si">%s</span><span class="s1">_2D.tif&#39;</span> <span class="o">%</span><span class="n">fov</span><span class="p">),</span> <span class="n">img</span><span class="p">)</span>
</pre></div>
</div>
</section>
</section>
<section id="step-2-watershed-segmentation-of-nucleus">
<h2>Step 2 Watershed segmentation of nucleus<a class="headerlink" href="#step-2-watershed-segmentation-of-nucleus" title="Link to this heading">¶</a></h2>
<p>The initial nucleus segmentation is required for refined cell segmentation by Cellist. In Cellist, we utilize the watershed algorithm to segment nuclei in the DAPI staining image, which is implemented by the function of <code class="code highlight bash docutils literal highlight-bash">watershed</code>. Here we take <cite>FOV_0</cite> as an example.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cellist<span class="w"> </span>watershed<span class="w"> </span>--platform<span class="w"> </span>imaging<span class="w"> </span><span class="se">\</span>
--gem<span class="w"> </span>Result/Preprocess/FOV_0/seqFISH+_NIH3T3_points_RNA_rep1_FOV_0_bin5.txt<span class="w"> </span><span class="se">\</span>
--tif<span class="w"> </span>Result/Preprocess/FOV_0/MMStack_Pos0_2D.tif<span class="w"> </span><span class="se">\</span>
--min-distance<span class="w"> </span><span class="m">100</span><span class="w"> </span><span class="se">\</span>
--outdir<span class="w"> </span>Result/Watershed/FOV_0<span class="w"> </span><span class="se">\</span>
--outprefix<span class="w"> </span>FOV_0
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/FOV_0_cell_boundary.png"><img alt="../_images/FOV_0_cell_boundary.png" class="align-center" src="../_images/FOV_0_cell_boundary.png" style="width: 100%;" /></a>
</section>
<section id="step-3-cell-segmentation-by-cellist">
<h2>Step 3 Cell segmentation by Cellist<a class="headerlink" href="#step-3-cell-segmentation-by-cellist" title="Link to this heading">¶</a></h2>
<p>With nucleus segmentation completed, the next step is to expand the nucleus labels to include the cytoplasm, namely, cell segmentation. In cellist, we take both expression similarity and spatial proximity into consideration when assigning non-nucleus spots to labelled nuclei.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cellist<span class="w"> </span>seg<span class="w"> </span>--platform<span class="w"> </span>imaging<span class="w"> </span><span class="se">\</span>
--resolution<span class="w"> </span><span class="m">0</span>.1<span class="w"> </span><span class="se">\</span>
--gem<span class="w"> </span>Result/Preprocess/FOV_0/seqFISH+_NIH3T3_points_RNA_rep1_FOV_0_bin5.txt<span class="w"> </span><span class="se">\</span>
--spot-count-h5<span class="w"> </span>Result/Preprocess/FOV_0/seqFISH+_NIH3T3_points_RNA_rep1_FOV_0_bin5.h5<span class="w"> </span><span class="se">\</span>
--nuclei-prop<span class="w"> </span>Result/Watershed/FOV_0/FOV_0_watershed_nucleus_property.txt<span class="w"> </span><span class="se">\</span>
--nuclei-count-h5<span class="w"> </span>Result/Watershed/FOV_0/FOV_0_waterhsed_segmentation_cell_count.h5<span class="w"> </span><span class="se">\</span>
--watershed-seg<span class="w"> </span>Result/Watershed/FOV_0/FOV_0_watershed_nucleus_coord.txt<span class="w"> </span><span class="se">\</span>
--nworkers<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="se">\</span>
--cell-diameter<span class="w"> </span><span class="m">18</span><span class="w"> </span><span class="se">\</span>
--spot-imputation-distance<span class="w"> </span><span class="m">2</span>.5<span class="w"> </span><span class="se">\</span>
--prob-cutoff<span class="w"> </span><span class="m">0</span>.1<span class="w"> </span><span class="se">\</span>
--outdir<span class="w"> </span>Result/Cellist/FOV_0<span class="w"> </span><span class="se">\</span>
--outprefix<span class="w"> </span>FOV_0
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/FOV_0_Cellist_segmentation_plot.png"><img alt="../_images/FOV_0_Cellist_segmentation_plot.png" class="align-center" src="../_images/FOV_0_Cellist_segmentation_plot.png" style="width: 100%;" /></a>
</section>
</section>


          </div>
          <div class="page-nav">
            <div class="inner"><ul class="page-nav">
  <li class="prev">
    <a href="Stereo-seq.html"
       title="previous chapter">← Mouse olfactory bulb (Stereo-seq)</a>
  </li>
  <li class="next">
    <a href="../release_notes/index.html"
       title="next chapter">Release notes →</a>
  </li>
</ul><div class="footer" role="contentinfo">
      &#169; Copyright 2024, Wang Lab at Tongji.
    <br>
    Created using <a href="http://sphinx-doc.org/">Sphinx</a> 7.3.5 with <a href="https://github.com/schettino72/sphinx_press_theme">Press Theme</a> 0.9.1.
</div>
            </div>
          </div>
      </page>
    </div></div>
    
    
  </body>
</html>