# Cellist
Cell identification in high-resolution Spatial Transcriptomics

Cellist is a computational method to perform cell segmentation on high-resolution spatial transcriptomics (ST) data, including sequncing-based (e.g. Stereo-seq and Seq-Scope) and imaging-based (e.g. seqFISH+ and STARmap) technologies.

<<<<<<< HEAD
![avatar](docs/_static/img/Cellist_workflow.png)
=======
![avatar](docs/_static/img/Cellist_framework.png)
>>>>>>> 1e8bbff2d5030abf814122911c412f811076177e

## Change Log
### v0.0.1a
* Build Cellist.
<<<<<<< HEAD
### v1.0.0
* Release Cellist.

=======
>>>>>>> 1e8bbff2d5030abf814122911c412f811076177e

## Install Cellist
```bash
git clone https://github.com/wanglabtongji/Cellist.git
cd Cellist
conda create -n Cellist python=3.10
pip install -r requirements.txt
<<<<<<< HEAD
pip install .
```

## Documentation
For full installation and usage of Cellist, please refer to the [documentation](https://cellist.readthedocs.io/en/latest/).

## Usage
```bash
cellist --help
usage: cellist [-h] [-v] {seg,align,watershed,impute} ...

Cellist (Cell identification in high-resolution Spatial Transcriptomics) is a cell segmentation tool for high-resolution spatial transcriptomics.

positional arguments:
  {seg,align,watershed,impute}
    seg                 Run Cellist segmentation on high-resolution spatial transcriptomics.
    align               Refine alignment between image and spatial transcriptomics.
    watershed           Run initial watershed segmentation on the staining image.
    impute              Perform spatially-aware gene imputation within each cluster.

optional arguments:
  -h, --help            show this help message and exit
  -v, --version         Print version info.
```
=======
python setup.py install
```

## Usage
```bash
cellist seg --help
usage: cellist seg [-h] [--gem SPOT_EXPR_FILE] [--spot-count-h5 ALL_SPOT_COUNT_H5_FILE] [--nuclei-prop PROPS_FILE] [--nuclei-count-h5 NUCLEUS_COUNT_H5_FILE]
                   [--watershed-seg WATERSHED_COORD_FILE] [--nworkers NUM_WORKERS] [--patch-data-dir PATCH_DATA_DIR] [--cell-diameter MAX_DIST]
                   [--spot-imputation-distance NEIGH_DIST] [--prob-cutoff PROB_CUTOFF] [--alpha ALPHA] [--sigma SIGMA] [--beta BETA] [--gene-use {HVG,Frequent}] [--iteration]

optional arguments:
  -h, --help            show this help message and exit

Input arguments:
  --gem SPOT_EXPR_FILE  Bin1 gem file generated by SAW pipeline.
  --spot-count-h5 ALL_SPOT_COUNT_H5_FILE
                        Bin1 spot-level expression file in h5 format.
  --nuclei-prop PROPS_FILE
                        Properties of segmented nucleus by Watershed algorithm.
  --nuclei-count-h5 NUCLEUS_COUNT_H5_FILE
                        Nuclei-level expression generated by Watershed segmentation with h5 format.
  --watershed-seg WATERSHED_COORD_FILE
                        Spot-level Watershed segmentation result.
  --nworkers NUM_WORKERS
                        Maximum number of workers to use.
  --patch-data-dir PATCH_DATA_DIR
                        Directory of patch data generated by Cellist (not used in the first run of Cellist).

Model arguments:
  --cell-diameter MAX_DIST
                        Expected cell diameter. Default: 15
  --spot-imputation-distance NEIGH_DIST
                        Imputation distance for each spot. Default: 5.0.
  --prob-cutoff PROB_CUTOFF
                        Cutoff for assignment probability. Default: 0.7.
  --alpha ALPHA         Weight for expression distance. Default: 0.8.
  --sigma SIGMA         Weight for regularization. Default: 1.0.
  --beta BETA           Penalty weight. Default: 10.
  --gene-use {HVG,Frequent}
                        Gene to use to calculate spot-cell distance. 'HVG' means highly variable genes, and 'Frequent' means top frequent genes. Default: HVG.
  --iteration           Iteration or not. If set, Cellist will assign non-nuclei spots iteratively with cell expression and centroid updated.
```
>>>>>>> 1e8bbff2d5030abf814122911c412f811076177e
